src/Components/Facility/Consultations/components/SharedTable.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { formatDate, formatTime } from "../../../../Utils/utils";
+import { classNames } from "../../../../Utils/utils";
+import { useTranslation } from "react-i18next";
+
+interface SharedSectionTableProps {
+  data: Record<string, any>;
+  rows: Array<{ title?: string; field?: string; subField?: boolean }>;
+  choices?: Record<string, Record<number | string, string>>;
+}
+
+const LogUpdateAnalayseTable: React.FC<SharedSectionTableProps> = ({
+  data,
+  rows,
+  choices = {},
+}) => {
+  const { t } = useTranslation();
+
+  // Helper function to get the display value
+  const getDisplayValue = (value: any, field?: string): string => {
+    if (value == null) {
+      return " ";
+    }
+
+    if (typeof value === "boolean") {
+      return t(value ? "yes" : "no");
+    }
+    if (field && choices[field]) {
+      const choice =
+        choices[field][value as keyof (typeof choices)[typeof field]];
+      return choice ? t(`${field.toUpperCase()}__${choice}`) : "-";
+    }
+    if (value && typeof value == "string") return value;
+
+    return "-";
+  };
+
+  const isddm_mm = (str: string) => {
```

Comment:
please use a descriptive function name, as lets stick to camelCasing for function names